### ISSUE TITLE
Max Batch Size is 6 on a 24GB 3090

### ISSUE THREAD

**Issue by DamonianoStudios** [derrian-distro/LoRA_Easy_Training_Scripts#14]
When running Kohya's official repo or the bmaltais GUI version, I am able to run a Batch size of 36 on my 3090, but if I go above 6 I get an OOM error when training with these scripts.  Even at BS 6 is uses 23.8 GB of vRAM, which seems absurd and goes against what you say of a 12gb card supporting BS 6.  I will attach some screenshots showcasing the vram usage when training with these scripts.
This training turned out bad but I blame my own settings as well as the fact I couldn't do my normal BS.
![image](https://user-images.githubusercontent.com/111029881/220439334-0ac8997d-f713-4ed8-9128-6659e1224c83.png)
![image](https://user-images.githubusercontent.com/111029881/220439371-964dc724-a89c-4352-a86e-8c38091d97e9.png)

**Comment by derrian-distro** [comment]
It is absurd, because I can get batch size 6 at 512 with my 12gb card, and usually use batch 6 when training at 512. I also know that my scripts don't impact the performance here because it just uses the same exact backend. This seems like an issue with your setup, perhaps duplicated runtime or something like that, but I have had many reports contrary to what you are reporting here.

Keep in mind, again, *my scripts uses kohya's SD-Scripts as the backend*, so the must lie in what else you had running at the time.

**Comment by DamonianoStudios** [comment]
What do you need for me to prove it is not my setup and an issue with how you are handling 30xx and probably 40xx cards? I can show you all my running processes while I have it open and it will still take 23.8 gb of VRAM, you are not capping the amount of memory for it to use.  No matter the BS it uses max memory

**Comment by derrian-distro** [comment]
I literally make no distinction between 30 and 40 series cards, I *don't handle the training* I handle inputting of variables, it is an issue with your system, as it uses the *exact same system* as both the kohya system and the ui

**Comment by DamonianoStudios** [comment]
Then why can I run the same backend using bmaltais GUI with a batch size of 36?

**Comment by derrian-distro** [comment]
I don't know why, I have never had an issue with batch size, and am able to get batch size 6 every time I run it, which is often, because I test everything before I release an update. I am not at my computer so I can't show you that it works fine on my machine, and many other people's machines, but this can only be related to your setup,

That being said, what resolution were you training at? Because the batch size metric is only for resolution 512

**Comment by DamonianoStudios** [comment]
I was training at 768x768, BUT that is the size I use when using kohya_ss command line or Bmaltais GUI, so it not working here tells me you didn't implement some sort of optimizations, or the popup ui is still hiding settings like gradient checkpointing behind a non obvious popup.  I just want to iterate, I like your scripts, but am trying to help increase batch size, since the other GUI or kohya's direct implementation allow for much higher batch size with the right settings.

**Comment by derrian-distro** [comment]
Then you must have been using gradient accumulation that isn't part of the asked questions. If you want to use that to increase your batch size you can, I have everything that was released in sd-scripts, as it just goes on top of it

**Comment by DamonianoStudios** [comment]
Alright so sounds like the pop-up ui just doesn't support that setting and I will need to use the command line script and have that set.  Thanks for your help, I can go ahead and close this issue since it is not really a bug.

**Comment by derrian-distro** [comment]
Alright, I didn't opt to include it because it is a somewhat confusing option, as it plays with your actual batch size. Command line is what you want to use if you want full control anyways, as popup is designed for people who were completely new, thay being said, command_line is also very easy to use.

That being said, apologies if I sounded annoyed, I was rushed for time because I am currently at work.
